### ISSUE TITLE
feature request: support APA format for "almost published" work using status field

### ISSUE THREAD

**Issue by kleinschmidt** [citation-style-language/styles#3183]
Biblatex supports a `pubstate` field which is used to indicate if a work is submitted, accepted, in press, etc.  `pandoc-citeproc` converts this to a `status` field in the JSON/YAML output that it produces.  There are three kinds of "almost published" works that the APA format refers to (e.g., http://blog.apastyle.org/apastyle/2012/08/almost-published.html):

* In preparation
* Submitted
* Advance online publication

I often want to refer to papers that are accepted but not published online yet (since there can be a substantial delay there), so it would be nice to support that, too.

To support these, the apa.csl file would need to check the status field and change the way the issued date and the journal, volume etc.  I might be able to manage that myself but wanted to open a discussion to see if a PR to that effect would be considered.

**Comment by kleinschmidt** [comment]
Also, the `status` field is mentioned in the official list of fields, with "forthcoming" as an example value, so presumably something like this is actually used in other formats.

**Comment by kleinschmidt** [comment]
Ah, I see that this is implemented in the zotero APA CSL file: http://editor.citationstyles.org/styleInfo/?styleId=http%3A%2F%2Fwww.zotero.org%2Fstyles%2Fapa

Is there any reason why it's not implemented here?

**Comment by rmzelle** [comment]
> To support these, the apa.csl file would need to check the status field and change the way the issued date and the journal, volume etc.

We can probably improve support in CSL a little bit (e.g. with a controlled vocabulary for the "status" variable), but of the three statuses you list, "In preparation", "Submitted", and "Advance online publication", you would typically only cite items with the last status in your bibliographies. Currently, we assume that authors just write out references to "In preparation" and "Submitted" items by hand in the text.

To determine if an article is "Advance online publication", CSL styles often check if the item is missing an issue or page numbers. (@adam3smith, correct me if I'm wrong here)

> Ah, I see that this is implemented in the zotero APA CSL file: http://editor.citationstyles.org/styleInfo/?styleId=http%3A%2F%2Fwww.zotero.org%2Fstyles%2Fapa
> 
> Is there any reason why it's not implemented here?

That's the style from this repository (I recognized that it's a bit confusing that all CSL repository styles have style IDs with the Zotero domain in them, but that's just how things developed historically).

**Comment by bwiernik** [comment]
@rmzelle No, in psychology and management journals, it’s common to cite in preparation and submitted manuscripts. But those get cited as `manuscript` not `article-journal`. From APA perspective, there are two statuses that should be supported: “in press” and “Advance online publication”. It would be good for these two to be implanted as terms (e.g., NLM and Chicago recommend “forthcoming” instead of in press).

**Comment by bwiernik** [comment]
@kleinschmidt The first two examples at the blog post you link to would be entered as Manuscript with the “status” in the Manuscript Type field. CSL type `manuscript` and CSL variable `genre`

**Comment by bwiernik** [comment]
@rmzelle @adam3smith I've been thinking about how to improve the use of the `status` variable and related terms (forthcoming, in press, etc.) in various styles. I think that most use cases can be covered with a controlled vocabulary with three options:

1. `forthcoming`: Indicating that an item is accepted for publication and somewhere in the production process. This status would call the terms `in press` or `forthcoming` as specified by the style.
1. `advance-online-publication`: Indicating that the item has been published online but not yet assigned a final volume/issue/page ("ahead of print"). This status would call a new term "advance online publication".
1. User-supplied text, to be printed literally.

**Comment by njbart** [comment]
@bwiernik If in preparation and submitted manuscripts are actually cited in some journals, shouldn’t these terms be included in the controlled vocabulary?

**Comment by bwiernik** [comment]
Those sorts of things are typically stored in the `genre` CSL variable with manuscript item types.

**Comment by clbarnes** [comment]
> advance-online-publication: Indicating that the item has been published online but not yet assigned a final volume/issue/page ("ahead of print"). This status would call a new term advance online publication.

I'm not a big fan of this because it implies that publications aren't online, and/or that online publications aren't real. In science, at least, referencing dead trees is becoming less and less meaningful. I appreciate that most citation machinery has a lot of history behind it, but there's no reason to intentionally put ourselves in a box where we pretend the internet isn't fundamental to the way that people access citeable works. `advance-publication` is fine; there's no reason to specify "online".

**Comment by adam3smith** [comment]
CSL tries to be mostly policy agnostic (there is no rational justification for the existence of most citation styles in the first place, after all...). The publishing system uses a lot of old-fashioned terminology (cf. also "preprint"), but I don't think the place to change that is in the syntax of a programming language. "Advance online publication" is very widely used, to using that at the attribute just seems like good literate programming practice to me.
